controllers/volumereplicationgroup_controller.go

```diff
@@ -718,6 +737,26 @@ func (v *VRGInstance) updateReplicationClassList() error {
 	return nil
 }
 
+func (v *VRGInstance) updateGroupReplicationClassList() error {
+	labelSelector := v.instance.Spec.Async.ReplicationClassSelector
+
+	v.log.Info("Fetching VolumeGroupReplicationClass", "labeled", labels.Set(labelSelector.MatchLabels))
+	listOptions := []client.ListOption{
+		client.MatchingLabels(labelSelector.MatchLabels),
+	}
+
+	if err := v.reconciler.List(v.ctx, v.grpReplClassList, listOptions...); err != nil {
+		v.log.Error(err, "Failed to list Group Replication Classes",
+			"labeled", labels.Set(labelSelector.MatchLabels))
+
+		return fmt.Errorf("failed to list Group Replication Classes, %w", err)
+	}
+
+	v.log.Info("Number of Group Replication Classes", "count", len(v.grpReplClassList.Items))
+
+	return nil
+}
+
```

Comment:
@ELENAGER 
Can you please match this update function with the changes coming from https://github.com/RamenDR/ramen/pull/1476/files#diff-a3c633f01e052ce1a42ea903ff76debacd956585919969d1f164ff88a7028f44 ? 

and are we sure we should be using the same `replicationClassSelector` varibale for groupReplicationClass too or have a separate `groupReplicationClassSelector` ? 

cc @ShyamsundarR  @Madhu-1 @kseegerrh